File: Taller2SGOSmestre/src/taller2sgosmestre/ListaClientes.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package taller2sgosmestre;

/**
 *
 * @author ACER
 */
public class ListaClientes {
    private int max;
    private int cant;
    private Cliente [] lc;

    public ListaClientes(int max) {
        this.max = max;
        this.cant = 0;
        this.lc = new Cliente[max];
    }

    public int getMax() {
        return max;
    }

    public void setMax(int max) {
        this.max = max;
    }

    public int getCant() {
        return cant;
    }

    public void setCant(int cant) {
        this.cant = cant;
    }

    public Cliente getCliente(int i) {
        return lc[i];
    }

    public void setLc(Cliente[] lc) {
        this.lc = lc;
    }
    public boolean ingresar(Cliente C){
        if(cant>=max){
            return false;
        }else{
            lc[cant]=C;
            cant++;
            return true;
        }
    }
    public Cliente buscarCliente(String rut){
        int i;
        for(i = 0;i<cant;i++){
            if(lc[i].getRut().equalsIgnoreCase(rut)){
                break;
            }
        }
        if(i == cant){
            return null;
        }else{
            return lc[i];
        }
    }
    
    
}

File: Taller1ED_2019-2/ListaClientes.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   ListaClientes.cpp
 * Author: Juan Pablo Martinez
 * 
 * Created on 18 de septiembre de 2019, 17:58
 */

#include "ListaClientes.h"

ListaClientes::ListaClientes() {
    this->first = nullptr;
    this->last = nullptr;
}

NodoCliente* ListaClientes::getFirst() {
    return this->first;
}

NodoCliente* ListaClientes::getLast() {
    return this->last;
}

void ListaClientes::setFirst(NodoCliente* first) {
    this->first = first;
}

void ListaClientes::setLast(NodoCliente* last) {
    this->last = last;
}

void ListaClientes::ingresarCliente(Cliente* c) {
    NodoCliente *nuevo = new NodoCliente(c);
    if (first == nullptr) {
        first = nuevo;
        last = nuevo;
    } else {
        nuevo->setNext(first);
        first->setPrev(nuevo);
        first = nuevo;
    }
}

Cliente* ListaClientes::buscarCliente(string nombre) {
    NodoCliente *actual = first;
    Cliente *buscado = nullptr;
    while(actual != nullptr){
        if(actual->getCliente()->getNombre() == nombre){
            buscado= actual->getCliente();
            break;
        }
        actual=actual->getNext();
    }
    return buscado;
}

bool ListaClientes::eliminarCliente(string nombre) {
    NodoCliente *actual = first;
    while(actual != nullptr && actual->getCliente()->getNombre() != nombre){
        actual = actual->getNext();
    }
    NodoCliente *aux = nullptr;
    if(actual != nullptr){
        if(actual == first){
            aux = first;
            first = first->getNext();
        }else if(actual == last){
            aux = last;
            last = last->getPrev();
        }else{
            aux = actual;
            actual->getNext()->setPrev(actual->getPrev());
            actual->getPrev()->setNext(actual->getNext());
        }
        delete aux;
        return true;
    }else{
        return false;
    }
}


ListaClientes::~ListaClientes() {
}

